cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,204 @@
+        file_paths: list[Path] = [
+            sample_folder.joinpath(path)
+            for path in get_files_matching_pattern(directory=sample_folder, pattern="*.fastq.gz")
+        ]
+        LOG.debug(f"Found {len(file_paths)} fastq files for sample {sample_id}")
+        hk_version: Version = self.housekeeper_api.get_or_create_version(bundle_name=sample_id)
         fastq_paths_to_add: list[Path] = self.housekeeper_api.check_bundle_files(
-            file_paths=paths,
-            bundle_name=lims_sample_id,
+            file_paths=file_paths,
+            bundle_name=sample_id,
             last_version=hk_version,
             tags=HK_FASTQ_TAGS,
         )
+        LOG.debug(f"{len(fastq_paths_to_add)} samples are not yet in Housekeeper for {sample_id}")
         return fastq_paths_to_add
 
-    def curate_sample_folder(self, cust_name: str, force: bool, sample_folder: Path) -> None:
-        """Changes the name of the folder to the internal_id. If force is true replaces any previous folder."""
-        customer: Customer = self.status_db.get_customer_by_internal_id(
-            customer_internal_id=cust_name
-        )
-        customer_folder: Path = sample_folder.parent
-        sample: Sample = self.status_db.get_sample_by_customer_and_name(
-            customer_entry_id=[customer.id], sample_name=sample_folder.name
-        )
-        if (sample and not customer_folder.joinpath(sample.internal_id).exists()) or (
-            sample and force
-        ):
-            sample_folder.rename(customer_folder.joinpath(sample.internal_id))
-        elif not sample and not self.status_db.get_sample_by_internal_id(sample_folder.name):
-            raise Exception(
-                f"{sample_folder} is not a sample present in statusdb. Move or remove it to continue"
+    def _add_and_include_files_to_bundles(
+        self, fastq_paths: list[Path], lims_sample_id: str
+    ) -> None:
+        """Add the given fastq files to the the Housekeeper bundle."""
+        if self.dry_run:
+            LOG.info(f"No files will be added for sample {lims_sample_id} since this is a dry-run")
+            return
+        for path in fastq_paths:
+            LOG.info(f"Adding path {path} to bundle {lims_sample_id} in housekeeper")
+            self.housekeeper_api.add_and_include_file_to_latest_version(
+                bundle_name=lims_sample_id, file=path, tags=HK_FASTQ_TAGS
             )
 
+    def _start_cases(self, cases: list[Case]) -> None:
+        """Starts the cases that have not been analysed."""
+        if self.dry_run:
+            LOG.info("No cases will be started since this is a dry-run")
+            return
+        for case in cases:
+            self.status_db.set_case_action(
+                case_internal_id=case.internal_id, action=CaseActions.ANALYZE
+            )
+            LOG.info(f"Case {case.internal_id} has been set to '{CaseActions.ANALYZE}'")
+
     def add_transfer_to_housekeeper(
```

Comment:
I am not a fan of this method name. I do not understand what a "transfer" is in this case, and think the added docstring indicates a violation of the SRP. If it both adds and includes fastq files and starts cases, maybe we should consider having two methods. One for adding/including files and one for setting the case actions.

Reply:
I renamed it to `add_external_data_to_housekeeper`. However modifying the flow for having two functions goes out of the scope of this PR I think, and should be discussed more carefully.